index.js

```diff
@@ -1,127 +1,115 @@
-const avilableServices = {};
 const { nanoid } = require('nanoid');
+const express = require('express');
 
 class Reshuffle {
   constructor() {
-    this.servicesIdToServices = {};
-    this.eventIdsToHandlers = {};
-    this.shareResources = {};
-    this.httpDelegetes = {};
-    this.webserverStarted = false;
-    this.port =8000;
-    console.log("Initiating Reshuffle");
+    this.registry = {
+      services : {},
+      handlers : {},
+      common : {}
+    };
+    this.httpDelegates = {};
+    this.port = process.env.RESHUFFLE_PORT || 8000;
 
+    console.log('Initializing Reshuffle');
+  }
+
+  createWebServer() {
+    this.registry.common.webserver = express();
+    this.registry.common.webserver.route('*')
+        .all((req, res, next) => {
+          if (this.httpDelegates[req.url]) {
+            this.httpDelegates[req.url].handle(req, res, next);
+          } else {
+            res.end(`No handler registered for ${req.url}`);
+          }
+        });
   }
   
-  use(service, service_id) {
+  register(service) {
     service.app = this;
-    if(service_id){
-      service.id = service_id;
-    }
-    this.servicesIdToServices[service.id] = service;
-     
+    this.registry.services[service.id] = service;  
   }
 
-  unuse(service,) {
-    service.stop()
-    delete this.servicesIdToServices[service.id];
+  async unregister(service) {
+    await service.stop()
+    delete this.registry.services[service.id];
   }
 
   getService(serviceId) {
-    return this.servicesIdToServices[serviceId];
+    return this.registry.services[serviceId];
   }
 
-  // eslint-disable-next-line no-unused-vars
-  unregisterHTTPDelegate(path, delegate) {
-    delete this.httpDelegetes[path];
-  }
   registerHTTPDelegate(path, delegate) {
-    if (!this.shareResources.webserver) {
-      var express = require('express');
-      this.shareResources.webserver = express();
-      this.shareResources.webserver.route("*")
-        .all((function (req, res, next) {
-          let handled = false;
-          if (this.httpDelegetes[req.url]) {
-            handled = this.httpDelegetes[req.url].handle(req, res, next);
```

Comment:
getting the true/false from the delegate is important, the edge case is that a script is removed from a path - the delegate is still there and no one returns an HTTP response on the socket.

Reply:
okay, I'll revert those